lib/buffy/throttle_and_timed.ex

```diff
@@ -203,11 +203,6 @@ defmodule Buffy.ThrottleAndTimed do
     throttle = Keyword.fetch!(opts, :throttle)
     loop_interval = Keyword.fetch!(opts, :loop_interval)
 
-    unless is_number(loop_interval) do
```

Comment:
`wms-service` dialyzer kept saying:

```
(ArgumentError) expected :loop_interval to be a number, received: {{:., [line: 9], [{:__aliases__, [counter: {WarehouseEvents.FulfillableStatusCheckBuffer, 1}, line: 9], [:Application]}, :get_env]}, [line: 9], [:warehouse, :fulfillable_status_check_loop_interval, 120000]}
    (buffy 2.1.0) expanding macro: Buffy.ThrottleAndTimed.__using__/1
    lib/warehouse_events/fulfillable_status_check_buffer.ex:2: WarehouseEvents.FulfillableStatusCheckBuffer (module)
    (elixir 1.15.4) expanding macro: Kernel.use/2
    lib/warehouse_events/fulfillable_status_check_buffer.ex:2: WarehouseEvents.FulfillableStatusCheckBuffer (module)
```

I think it has to do with compile time vs run time code. 
Instead of dealing with type checking here, I tested by giving it a binary, and the dialyzer correctly gave an error during parsing for its use in `handle_continue` - so I think we can do without this explicit check.